torch/csrc/distributed/rpc/tensorpipe_agent.h

```diff
@@ -319,11 +324,16 @@ class TensorPipeAgent : public RpcAgent {
   // Map to store the expiration times for each message.
   std::map<
       steady_clock_time_point,
-      std::vector<std::pair<
+      std::vector<std::tuple<
+          uint64_t, // messageId
```

Comment:
Nit, and unrelated to this diff: Now that we have messageId here, we can probably remove the `std::shared_ptr<AtomicJitFuture>` key from this map and just use the messageId to key into the pending futures map. This is similar to what process group agent does.